mdoc/src/main/scala-2/mdoc/internal/markdown/Renderer.scala

```diff
@@ -92,6 +93,23 @@ object Renderer {
     }
   }
 
+  def appendMultilineMaxWidth(sb: PrintStream, string: String, maxLineLength: Int): Unit = {
+    var i = 0
+    var lineLength = 0
+
+    while (i < string.length()) {
+      string.charAt(i) match {
+        case '\n' =>
+          sb.append("\n// "); lineLength = 0
+        case ch if lineLength == maxLineLength =>
+          sb.append(ch); sb.append("\n// "); lineLength = 0
```

Comment:
I'm a bit skeptical about this line wrapping as it breaks the output at arbitrary offsets. I think there is a risk you line wrap right before a newline like this
```scala
List(
  "aaaaa
",
  "bbb"
)
```

I feel like it would be better to make this line behavior opt-in with an additional modifier `mdoc:wrap-width=anywhere` where the default value is something like `wrap-width=tokens` or `wrap-width=words`. What do you think?

Reply:
I can imagine there are cases where users would like to set a smaller width while keeping the current line wrapping behavior (split at tokens/words)

Reply:
I was thinking about it, yeah. Width is the problematic one as in the terminal you would expect the behaviour you showed in your example, but in markdown it might look weird.

I need to look closer at what pprint exposes (token-wise) as it indeed would be nicer to break on tokens only, but being able to set `anywhere` is even better to break long strings (only situation where I can imagine this being useful)

Reply:
I don't think pprint offers line breaking beyond tokens. I don't mind exposing the ability to break inside tokens but I think it should be an explicit choice from the user

Reply:
I had a complete brain fart moment :) I struggled to find an example of demonstrating that pprint actually does wrap at given width, around tokens.

````

```scala mdoc:width=50
List.fill(10)(List(1,2,3,4,5,6,7,8,9,10))
```

```scala
List.fill(10)(List(1,2,3,4,5,6,7,8,9,10))
// res0: List[List[Int]] = List(
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
//   List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
// )
```




```scala mdoc:width=20
List.fill(10)(List(1,2,3,4,5,6,7,8,9,10))
```

```scala
List.fill(10)(List(1,2,3,4,5,6,7,8,9,10))
// res1: List[List[Int]] = List(
//   List(
//     1,
//     2,
//     3,
//     4,
//     5,
//     6,
//     7,
//     8,
//     9,
//     10
//   ),
//   List(
//     1,
//     2,
//     3,
//     4,
//     5,
//     6,
//     7,
//     8,
//     9,
//     10
//   ),
//   List(
//     1,
//     2,
//     3,
//     4,
//     5,
//     6,
//     7,
//     8,
//     9,
//     10
//   ),
//   List(
//     1,
//     2,
//     3,
//     4,
//     5,
//     6,
//     7,
//     8,
//     9,
//     10
//   ),
// ...
```
````

which shows that it works (and I don't need to do manual breaking).

For my usage I don't actually need this "break anywhere" behaviour because it's unpredictable and can lead to bad rendering.

So unless you explicitly want to keep it in (guarded by a custom modifier), I will remove this function altogether and modify the tests - how does that sound?